R/filter_query_parameters.R

```diff
@@ -1,33 +1,32 @@
 #' Remove or replace query parameters
 #' @noRd
-query_params_remove <- function(int) {
-  h <- vcr_c$filter_query_parameters
-  if (!is.null(h)) {
-    if (is.null(names(h))) toremove <- unlist(h)
-    if (!is.null(names(h))) toremove <- unname(unlist(h[!nzchar(names(h))]))
-    # remove zero length strings
-    toremove <- Filter(nzchar, toremove)
-    for (i in seq_along(toremove)) {
-      int <- lapply(int, function(b) {
-        b$request$uri <- drop_param(b$request$uri, toremove[i])
-        return(b)
-      })
-    }
+query_params_remove <- function(
+  uri,
+  filter = vcr_c$filter_query_parameters,
+  flip = FALSE
+) {
+  is_named <- names2(filter) != ""
 
-    toreplace <- h[nzchar(names(h))]
-    if (length(toreplace)) {
-      for (i in seq_along(toreplace)) {
-        int <- lapply(int, function(b) {
-          vals <- toreplace[[i]]
-          val <- if (length(vals) == 2) vals[2] else vals[1]
-          b$request$uri <-
-            replace_param(b$request$uri, names(toreplace)[i], val)
-          return(b)
-        })
+  to_remove <- filter[!is_named]
+  for (i in seq_along(to_remove)) {
+    uri <- drop_param(uri, to_remove[[i]])
+  }
+
+  to_replace <- filter[is_named]
+  for (i in seq_along(to_replace)) {
+    val <- to_replace[[i]]
+    if (length(val) == 2) {
+      if (flip) {
```

Comment:
I don't understand why this is needed, but the tests, as well as careful comparison of `query_params_remove()` and `query_params_remove_str()` revealed that it's necessary.

Reply:
don't remember, I'll have a look